apis/tests.py

```diff
@@ -166,3 +169,294 @@ def test_vote_without_area(self):
         self.assertFalse(VoteCheck.objects.filter(user=self.users[0], election=self.election).exists())
 
 
+class CandidateApiTest(APITestCase):
+    def setUp(self):
+        """Pre-test setup."""
+        staff = User.objects.create_user(username="staff", email="staff@email.com", password="AmongUs")
+        staff.is_staff = True
+        staff.save()
+
+        self.users = [
+            # Staff
+            staff,
+
+            # Candidates
+            User.objects.create_user(username="U1", email="pp1@email.com", password="AmongUs"),
+            User.objects.create_user(username="U2", email="pp2@email.com", password="AmongUs"),
+            User.objects.create_user(username="U3", email="pp3@email.com", password="AmongUs")
+        ]
+
+        self.areas = [
+            NewArea.objects.create(name="A1"),
+            NewArea.objects.create(name="A2"),
+            NewArea.objects.create(name="A3")
+        ]
+
+        self.candidates = [
+            NewCandidate.objects.create(user=self.users[1], area=self.areas[0]),
+            NewCandidate.objects.create(user=self.users[2], area=self.areas[1]),
+            NewCandidate.objects.create(user=self.users[3], area=self.areas[2])
+        ]
+
+        self.url = reverse('api_candidate_list')
+
+    def test_get_has_candidates(self):
+        """All available candidates should be returned."""
+
+        response = self.client.get(self.url, format="json")
+        response_content = json.loads(response.content.decode("utf-8"))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response_content["result"]), 3)
+
+    def test_post_valid(self):
+        """Happy path test on adding a candidate. It should pass no matter what."""
+        test_user_id = User.objects.create_user(username="U4", email="pp4@email.com", password="AmongUs").id
+        test_area_id = self.areas[2].id
+        description = "Bla Bla Bla"
+
+        self.client.force_login(self.users[0])
+        response = self.client.post(self.url, {
+            "user_id": test_user_id,
+            "description": description,
+            "area_id": test_area_id
+        })
+        response_content = json.loads(response.content.decode("utf-8"))
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response_content["result"]["user"]["id"], test_user_id)
+        self.assertEqual(response_content["result"]["description"], description)
+        self.assertEqual(response_content["result"]["area"]["id"], test_area_id)
+
+        # Check created candidate's existence.
+        candidate_exists = True
+        try:
+            NewCandidate.objects.get(id=response_content["result"]["id"])
+        except ObjectDoesNotExist:
+            candidate_exists = False
+        self.assertTrue(candidate_exists)
```

Comment:
Django has `exists()` that check existence. The whole block can be reduced to.

```suggestion
self.assertTrue(NewCandidate.objects.filter(id=response_content["result"]["id"]).exists())
```

Reply:
Ah I see, I've modified my test as suggested.